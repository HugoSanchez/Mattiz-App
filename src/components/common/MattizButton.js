import React from 'react';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import LinearGradient from 'react-native-linear-gradient';

// Props:
// @containerStyles: -Optional- Wrapping container styles.
// @title: -Required- Text to be displayed.
// @titleStyle: - Optional - Text style for the title - there is a default.
// @buttonStyle: -Optionsl - Additional button styles.
// @onPress: -Required- Function to be executed on press. 
// @linearColor: -Optional - RGBA value for second linear color. Default: Mattiz Blue.

const MattizButton = props => {
    const { container, titleStyle, buttonStyle } = styles;
    return (
        <View style={[ container, props.containerStyles ]}>
            <Button 
                title={props.title}
                titleStyle={[ titleStyle, props.titleStyle ]}
                buttonStyle={[ buttonStyle, props.buttonStyle ]}
                onPress={ props.onPress }
                ViewComponent={ LinearGradient }
                linearGradientProps={{
                    colors: ['rgba(4, 0, 38, 0.1)', props.linearColor || 'rgba(4, 0, 38, 0.1)'],
                    start: { x: 0, y: 0.5 },
                    end: { x: 1, y: 0.5 },
                }}
            /> 
        </View>
    );
}

const styles = {
    container: {
        margin: 10, 
        borderRadius: 12,
        backgroundColor: '#FDFFFE',
        shadowColor: '#000',
        shadowOffset: { width: 2, height: 10 },
        shadowOpacity: 0.29,
        shadowRadius: 12.65,
        elevation: 22,
    },
    titleStyle: {
        color: 'white', 
        fontFamily: 'Raleway-Regular'
    },
    buttonStyle: {
        height: 60, 
        borderRadius: 10,
    }
}

export default MattizButton;