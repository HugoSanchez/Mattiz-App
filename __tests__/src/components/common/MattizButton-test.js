import 'react-native'
import React from 'react'
import MattizButton from '../../../../src/components/common/MattizButton'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<MattizButton />)
	expect(container.children[0]).toMatchSnapshot()
})
