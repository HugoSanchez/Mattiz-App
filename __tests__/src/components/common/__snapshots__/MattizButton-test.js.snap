// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFF",
        "borderRadius": 12,
        "elevation": 22,
        "margin": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 10,
          "width": 2,
        },
        "shadowOpacity": 0.29,
        "shadowRadius": 12.65,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.3}
      disabled={false}
    >
      <BVLinearGradient
        colors={
          Array [
            4292734455,
            4292734455,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 0.5,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 0.5,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 10,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "center",
            "padding": 8,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;
